Enhancements for building licenses for applications

bmcdavid opened the pull request:
added solution parsing support
fallback for failed nuspec requests
added project filter and allowable licenses
added license validation and validation result
added manual information, ability to include / disable project urls

bmcdavid commented:
Hi Tom,

I found your excellent project and needed to add some functionality to create license JSON files for my company's application. I figured I would share and hope you find them helpful. Feel free to ask me any questions.

Thanks,
Brad

tomchavakis commented on tests/NugetUtility.Tests/ProgramTests.cs:
```diff
@@ -0,0 +1,87 @@
+            Assert.AreEqual(1, status);
+        }
+
+        [TestCase("-i " + TestSetup.ThisProjectSolutionPath + @" --allowed-license-types ..\..\..\SampleAllowedLicenses.json")]
```
TestCase Fails on Linux due to Windows Folder Separator Structure

tomchavakis commented on tests/NugetUtility.Tests/ProgramTests.cs:
```diff
@@ -0,0 +1,87 @@
+        [Test]
+        public async Task Main_Should_Run_For_This_Project()
+        {
+            var status = await Program.Main(@"-i ..\..\..\".Split(' '));
```
same here

tomchavakis commented on tests/NugetUtility.Tests/TestSetup.cs:
```diff
@@ -0,0 +1,10 @@
+    [SetUpFixture]
+    public class  TestSetup
+    {
+        public const string ThisProjectSolutionPath = @"..\..\..\..\..\NugetUtility.sln";
```
Change to Path.DirectorySeparatorChar for Windows/Unix-based systems

tomchavakis commented on tests/NugetUtility.Tests/ProgramTests.cs:
```diff
@@ -0,0 +1,87 @@
+        [Test]
+        public async Task Main_Should_Run_For_This_Solution_With_Custom_Outfile_FullPath()
+        {
+            const string args = "-i " + TestSetup.ThisProjectSolutionPath + @" -j --outfile c:\temp\custom.json";
```
same here

tomchavakis commented on tests/NugetUtility.Tests/MethodsTests.cs:
```diff
@@ -0,0 +1,109 @@
+        {
+            var methods = new Methods(new PackageOptions
+            {
+                ProjectsFilterOption = @"..\..\..\SampleProjectFilters.json",
+                AllowedLicenseTypesOption = @"..\..\..\SampleAllowedLicenses.json",
```
same here

tomchavakis commented on tests/NugetUtility.Tests/MethodsTests.cs:
```diff
@@ -0,0 +1,109 @@
+        [SetUp]
+        public void Setup()
+        {
+            _projectPath = @"..\..\..\";
```
same here

tomchavakis commented:
Hi @bmcdavid ,

I would like to thank you for your awesome contribution !!!
I use Linux and VSCode and I had difficulty running the tests due to Windows path setup, but after altering them to the Unix-based path, runs successfully. 

Could you please change the README usage in order to be updated?
Would you mind adding your username as a contributor to this library? 

Thank's in advance

bmcdavid commented:
Hi @tomchavakis,

I've fixed the path issue to work for all environments and updated the readme. Also feel free to add my username as a contributor :).

Thanks,
Brad

bmcdavid commented:
@tomchavakis Any chance you can make a new NuGet package on NuGet.org with these changes?

Thanks in advance!

tomchavakis commented:
Hi @bmcdavid,
I have released the new version of nuget-license project.
You can find it at releases and as a Nuget package.
https://www.nuget.org/packages/dotnet-project-licenses/2.0.0

Thanks again for your contribution.
